Launchpad: Move tasklist view events to launchpad-internal

markbiek opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes #86970

## Proposed Changes

As part of restructuring/improving the Launchpad tracks events, this moves the mechanism for tracking Launchpad checklist views in the LaunchpadInternal component.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* This is easiest to test locally so you can see the tracks events in the dev console. But you could also use the calypso.live link and the Tracks Vigilante browser extension.
* Apply this branch locally and run calypso.
* Create a new site by visiting http://calypso.localhost:3000/setup/free
* When you get to the full screen Launchpad, look at the browser dev console.
* You should see one `calypso_launchpad_tasklist_viewed` event and lots of `calypso_launchpad_task_view` events.

![CleanShot 2024-01-30 at 14 29 53](https://github.com/Automattic/wp-calypso/assets/917632/671799ba-d2b1-4fbb-9a5b-431802a3de86)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

markbiek committed:
Move tasklist view events to launchpad-internal

gabrielcaires reviewed (changes requested):
Since tracking is very important for us, I suggest add a unit test just to protect us from future changes.

gabrielcaires commented:
Actually, I saw the context is always undefined during the onboarding, it should be fullscreen?

markbiek commented:
> Actually, I saw the context is always undefined during the onboarding, it should be fullscreen?

@gabrielcaires I'm not sure I understand what you mean. Can you give me any more context?

gabrielcaires commented:
> > Actually, I saw the context is always undefined during the onboarding, it should be fullscreen?
> 
> @gabrielcaires I'm not sure I understand what you mean. Can you give me any more context?

@markbiek When we open the customer home, the `calypso_launchpad_tasklist_viewed`  has the prop `context: customer-home` but when we open it on launchpad we don't have any context, I think it should be 'fullscreen' of something that helps us to identify the task was rendered during the onboarding flow.

markbiek commented:
> > > Actually, I saw the context is always undefined during the onboarding, it should be fullscreen?
> > 
> > 
> > @gabrielcaires I'm not sure I understand what you mean. Can you give me any more context?
> 
> @markbiek When we open the customer home, the `calypso_launchpad_tasklist_viewed` has the prop `context: customer-home` but when we open it on launchpad we don't have any context, I think it should be 'fullscreen' of something that helps us to identify the task was rendered during the onboarding flow.
> 
> Same on task_view events

Ah, I understand. I think that's a good idea but should be added in a separate diff after this one is approved.

I'm going to try and write that new test shortly so we can get this shipped.

sixhours commented:
I'm seeing the events, but they are firing twice; once when some properties are `undefined`, and then again with the correct event props:

<img width="1493" alt="Screenshot 2024-01-31 at 1 07 33 PM" src="https://github.com/Automattic/wp-calypso/assets/2124984/dc096f08-34db-4d55-8d96-b49833f330e7">
<img width="1484" alt="Screenshot 2024-01-31 at 1 07 55 PM" src="https://github.com/Automattic/wp-calypso/assets/2124984/6708142b-505c-4563-9568-5f21f38eadc0">

Going to check that that's not happening on `trunk`

markbiek committed:
Pass checklistSlug to LaunchpadInternal

The sidebar wasn't passing the checklistSlug so we sometimes fired Tracks events with an undefined value

markbiek commented:
@sixhours FYI, that issue with the `undefined` value should be fixed now.

sixhours reviewed (approved):
Looks good!
